iproute2: bump to 4.11

kdarbyshirebryant opened the pull request:
Signed-off-by: Kevin Darbyshire-Bryant <kevin@darbyshire-bryant.me.uk>

LynxChaus commented:
Did you test it?
source file hash not match: 22b1e1c1fc704ad35837e5a66103739727b8b48ac90b48c13f79b7367ff0a9a8 from 4.10.0, correct 72671028bda696d0cb8f48ec8e702581c3a501caeed33eec3a81d7041cbc8026,
patch 950-add-cake-to-tc.patch damaged, first hunk not apply (mismatched changed lines 61 <> 60).

hauke commented on package/network/utils/iproute2/Makefile:
```diff
@@ -49,7 +51,7 @@ endef
 $(call Package/iproute2/Default)
   TITLE:=Traffic control utility
-  DEPENDS:=+kmod-sched-core
+  DEPENDS:=+kmod-sched-core +libxtables +libmnl
```
libmnl and libxtbales are optional dependencies, which features do you need?

hauke commented on package/network/utils/iproute2/Makefile:
```diff
@@ -72,6 +74,12 @@ $(call Package/iproute2/Default)
+$(call Package/iproute2/Default)
+  TITLE:=Network devlink utility
+  DEPENDS:=+libmnl
+endef
```
yes for devlink you bee libmnl

hauke commented on package/network/utils/iproute2/patches/001-config.patch:
```diff
@@ -1,7 +1,8 @@
 +# Fixed config to disable ATM support even if present on host system
 +TC_CONFIG_ATM:=n
 +TC_CONFIG_ACTION_GACT=y
 +TC_CONFIG_ACTION_PROB=y
```
TC_CONFIG_ACTION_GACT=y and TC_CONFIG_ACTION_PROB=y do not exist any more.
Please add TC_CONFIG_XT:=y which would be added in case xtables is available

hauke commented on package/network/utils/iproute2/Makefile:
```diff
@@ -90,11 +98,12 @@ TARGET_CFLAGS += -DHAVE_SETNS
 TARGET_CFLAGS += -ffunction-sections -fdata-sections
```
please add IP_CONFIG_SETNS to Concifg file instead of manually activating HAVE_SETNS here. Normally the Config file gets generated with such information.

kdarbyshirebryant replied:
Done

hauke commented on package/network/utils/iproute2/patches/999-honor-copts.patch:
```diff
@@ -0,0 +1,11 @@
+ WFLAGS += -Wmissing-declarations -Wold-style-definition -Wformat=2
+ 
+-CFLAGS := $(WFLAGS) $(CCOPTS) -I../include $(DEFINES) $(CFLAGS)
++CFLAGS := $(WFLAGS) -I../include $(EXTRA_CCOPTS) $(DEFINES) $(CFLAGS)
```
please merge this together with 110-extra-ccopts.patch it does a very similar think in a different line.

kdarbyshirebryant replied:
Actually I think the entire 999 patch is bogus since 110-extra-ccopts does
CCOPTS = -O2 $(EXTRA_CCOPTS)

So CCOPTS already has $EXTRA_CCOPTS.

hauke commented on package/network/utils/iproute2/Makefile:
```diff
@@ -6,15 +6,17 @@
 #
 
 include $(TOPDIR)/rules.mk
+include $(INCLUDE_DIR)/kernel.mk
```
why is this needed?

kdarbyshirebryant replied:
No idea - I'll try removing it and see what happens.

kdarbyshirebryant commented:
Guys, thanks for the review.  The reality is that my $CLUE=NULL when it comes to iproute2.  Russell Senior had a bump to 4.10 that he was working on a month or two ago which since went quiet.  I wanted to make sure this work wasn't lost as it's a bit of a challenge getting it to compile with musl and see if I could get it to current level 4.11.  I had a number of little tweaks related to 'cake' in my own tree, hence the flustercluck with non applying patches and somehow the package HASH when I cherry-picked it as a single commit for this PR.  I'll update the PR addressing the points I understand once I can get the thing to compile again :-)

nbd168 commented:
Merged to my staging tree, thanks
